clang/include/clang/AST/Type.h

```diff
@@ -4699,7 +4699,7 @@ class FunctionEffect {
 
 private:
   LLVM_PREFERRED_TYPE(Kind)
-  unsigned FKind : 3;
+  uint8_t FKind : 3;
```

Comment:
The size of FunctionEffect changes from 4 (sizeof(unsigned)) to 1, and saves on memory footprint in the new analysis code.

Reply:
I'd split this into it's own NFC PR, and land it before the rest.

Reply:
also, github won't let me comment on the correct lines, but:

```
-  FunctionEffect() : FKind(unsigned(Kind::None)) {}
-  explicit FunctionEffect(Kind K) : FKind(unsigned(K)) {}
  
+  FunctionEffect() : FKind(uint8_t(Kind::None)) {}
+  explicit FunctionEffect(Kind K) : FKind(uint8_t(K)) {}
```

Reply:
(sorry, what's "NFC" in this context?)

Reply:
"no functional change"

https://llvm.org/docs/Lexicon.html#n

Reply:
Opened PR https://github.com/llvm/llvm-project/pull/100753 for this.

Reply:
That PR has been merged.